src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/remoteJob/BuildStatus.java

```diff
@@ -0,0 +1,92 @@
+package org.jenkinsci.plugins.ParameterizedRemoteTrigger.remoteJob;
+
+import hudson.model.Result;
+
+/**
+ * The build status of a remote build either reflecting a {@link hudson.model.Result} if finished,
+ * or if not finished yet a custom status like QUEUED, RUNNING,...<br>
+ * Using {@link #isJenkinsResult()} it can be checked if the status if reflecting a {@link Result} or a custom status.<br>
+ * The Jenkins {@link Result} can be obtained using {@link BuildStatus#getJenkinsResult()}.
+ */
+public enum BuildStatus
```

Comment:
Why do you need to clone Enum from the Jenkins core? If Jenkins project introduces new values, it may fall apart

Reply:
here the correct PR in our fork: https://github.com/sap-production/parameterized-remote-trigger-plugin/pull/23

Reply:
@oleg-nenashev @alejandraferreirovidal 
Our `BuildStatus` contains more information about the build, e.g. if it is still in queue, if it is already running, etc. `hudson.model.Result` only represents the result **after** the build finished.
This means by switching the `Result` we would loose valuable information. Users would not see if the remote build is running or not. In https://github.com/sap-production/parameterized-remote-trigger-plugin/pull/23 you can see which information would be lost.

I would vote for not switching to `hudson.model.Result`.